apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -60,248 +73,197 @@ export function OAuth2Form() {
-              // auto-consent!
-              onConsent()
-            } //else if prompt is defined and another value ('login', 'consent', or 'select_account') then always prompt!
-          } else if (prompt && prompt === 'none') {
-            // granted === false and prompt === none
-            onError(
-              new OAuthClientError(
-                'consent_required',
-                'Current user has not previously granted permission, and prompt was set to none',
-              ),
-            )
-          }
-        })
-        .catch((_err: Error | OAuthClientError | SynapseClientError) => {
-          onError(_err)
-        })
-        .finally(() => {
-          setIsPreviousAuthCheckComplete(true)
-        })
-    }
-    if (profile && !error) {
-      getHasAlreadyConsented()
-    }
-  }, [profile, error, accessToken, onConsent, onError])
-
-  const getOAuth2RequestDescription = () => {
-    if (!oidcRequestDescription && !error) {
-      const code = getURLParam('code')
-      if (code) return // we're in the middle of a SSO, do not attempt to get OAuth2RequestDescription yet
-
-      const request: OIDCAuthorizationRequest =
-        getOIDCAuthorizationRequestFromSearchParams()
-      SynapseClient.getOAuth2RequestDescription(request)
-        .then((oidcRequestDescription: OIDCAuthorizationRequestDescription) => {
-          if (isMounted.current) {
-            sendGTagEvent('SynapseOAuthClientRequestDescriptionLoaded')
-            setOidcRequestDescription(oidcRequestDescription)
-
-            // if we were able to get the oidc request description, also check for params that this web app does not support
-            // sorry, we don't support JWT in the url query params today
-            // https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests
-            const requestObject = getURLParam('request')
-            const requestUri = getURLParam('request_uri')
-            if (requestObject) {
-              handleErrorRedirect(new OAuthClientError('request_not_supported'))
-            }
-            if (requestUri) {
-              handleErrorRedirect(
-                new OAuthClientError('request_uri_not_supported'),
-              )
-            }
-            // sorry, we don't support registration (yet?)
-            // https://openid.net/specs/openid-connect-core-1_0.html#RegistrationParameter
-            const registration = getURLParam('registration')
-            if (registration) {
-              handleErrorRedirect(
-                new OAuthClientError('registration_not_supported'),
-              )
-            }
-          }
-        })
-        .catch(_err => {
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-          onError(_err)
-        })
-    }
+    // Redirect with 'access_denied' per https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1
+    handleErrorRedirect(new OAuthClientError('access_denied'))
```

Comment:
Let the `handleErrorRedirect` utility handle this case. Changed the behavior to send an `access_denied` error code to the client (as described in the OAuth2 spec)